tests/kernel/fpu_sharing/generic/CMakeLists.txt

```diff
@@ -13,5 +13,9 @@ project(fpu_sharing)
 set_ifndef(PI_NUM_ITERATIONS 700000)
 zephyr_compile_definitions(PI_NUM_ITERATIONS=${PI_NUM_ITERATIONS})
 
+if(DISABLE_INT_TEST)
+  zephyr_compile_definitions(DISABLE_INT_TEST=${DISABLE_INT_TEST})
+endif()
```

Comment:
This seems clumsy.  If FPU accessibility in interrupts is an arch-dependent thing (and it seems like it is), then you just want to test for architecture kconfigs in the C code.  Or if you want to be fancy, let's introduce a CONFIG_ARCH_HAS_FPU_IN_ISR kconfig that people can use everywhere.  Creating a CMake variable just for the purpose of defining a test-specific preprocessor symbol seems maybe a little silly?